src/Core/Utilities/ServiceCollectionExtensions.cs

```diff
@@ -46,49 +47,97 @@ public static class ServiceCollectionExtensions
     {
         public static void AddSqlServerRepositories(this IServiceCollection services, GlobalSettings globalSettings)
         {
-            var usePostgreSql = CoreHelpers.SettingHasValue(globalSettings.PostgreSql?.ConnectionString);
-            var useEf = usePostgreSql;
+            var selectedDatabaseProvider = globalSettings.DatabaseProvider;
```

Comment:
Would it make sense to throw an exception here if the provider was configured to something and the appropriate connection string was not provided?

Reply:
Yes. Done!